### ISSUE TITLE
Feature: Alert when an inline disable comment can be removed

### ISSUE THREAD

**Issue by hzoo** [eslint/eslint#5801]
Example:

```js
try {
  func();
} catch (e) {} // eslint-disable-line no-empty
```

This could be a separate command (say `eslint --check-disabled-comments`, or something crazier like after updating to a new version given an eslintrc). After running, it will iterate over every disable comment (like `// eslint-disable-line`, etc) and then try to try the file with the rule/rules specified and see if it doesn't error anymore. If it doesn't then it will tell the user that the comment can be removed now. If the rule has multiple options, then it can try each option and tell the user the option that passes.

As an example for https://github.com/eslint/eslint/issues/5800 - if we decide to add an exception for the empty catch block, then running the command would tell the user that you can remove the comment + change the rule to use the new option.

The point of this is to allow more of the consumers code to be linted by removing the disabled comments when possible.

**Comment by platinumazure** [comment]
Don't we already have a CLI option for ignoring inline ESLint config comments? If so, I don't know that we actually need this.

**Comment by hzoo** [comment]
What do you mean by ` ignoring inline ESLint config comments`?

I'm saying that the user has some disable comments currently in the code (due to an eslint error), but the comments could actually be removed either because the code was modified, eslint fixed a bug, or there was a new rule option that was added that would make the code pass without the comment.

**Comment by platinumazure** [comment]
I mean we have a CLI switch which ignores the inline disable comments and
other configuration comments.

However, I now see the problem: The report you get with that switch is the
exact opposite of what you want. You would see errors for those sections of
code with inline disable comments that actually have problems, and you
would see nothing for error-free code that still has inline comments. It
would be hard to find the comments that could be removed.

I could see some utility for the change, but I don't know if the
complexity/maintainability would necessarily be worth it.

**Comment by platinumazure** [comment]
@hzoo I have preliminarily labeled this as CLI (i.e., would be toggled by CLI option). But do you think there is value in having this available in the CLIEngine API? If so, I can switch the "cli" label to "core".

**Comment by alberto** [comment]
I think this should better be done as a custom tool

**Comment by ilyavolodin** [comment]
Agree with @alberto There has been a few requests for stuff like this (rules that can be enabled without any new errors, marking unused comments, etc.), and I don't think this fits into the core. It would be pretty easy to create a custom tool for this, though. Just run eslint once with `--no-inline-config` and once without and compare the output.
We wouldn't really have any better way of doing this in the core, so I don't think it's a good fit for the core.

**Comment by nzakas** [comment]
Agree with @alberto and @ilyavolodin, so closing.
